src/components/board/index.tsx

```diff
@@ -8,16 +8,71 @@ const Board: React.FC<BoardProps> = ({
 }) => {
     const [pieceState, setPieceState] = useState(pieceList)
 
+    const [selectedIndex, setSelectedIndex] = useState(-1)
+
+    const [points, setPoints] = useState(0);
+
+    const [canClick, setCanClick] = useState(true);
+
+    const [didWin, setDidWin] = useState(false);
+
+
+    useEffect(() => {
+        document.title = `Voce tem ${points} pontos`
+    }, [points]);
+
     const handleOnClick = (index: number) => {
-        pieceState[index].turned = !pieceState[index].turned
-        setPieceState([...pieceState])
+        if (canClick) {
+            pieceState[index].turned = !pieceState[index].turned
+            if (selectedIndex !== -1) {
+                if (pieceState[index].name === pieceState[selectedIndex].name) { 
+                    setCanClick(false)
+                    setTimeout(() => {
+                        setPoints(points + 10)
+                        pieceState[index].erased = true
+                        pieceState[selectedIndex].erased = true
+                        setPieceState([...pieceState])
+                        setCanClick(true)
+                        let pieceErased = true;
+                        for (let i = 0; i < pieceState.length; i++) {
```

Comment:
Quando puder opte por usar o forEach ao invés do for tradicional. No próximo treinamento vou mostrar o motivo.

Reply:
Também podemos usar um .some aqui para percorrer o array e ver se temos qualquer peça ainda não virada
`            
const hasPieceVisible = pieceState.some(p => !p.erased)
      if(!hasPieceVisible)
          setDidWin(true)
`

Reply:
ou alternativamente ter um estado de números que conta quantas peças foram apagadas até agora e caso seja o mesmo numero de peças do board é vitória, assim já nos livramos do laço